Use (only) MacPorts-installed scripts which are suffixed
with `${perl_version}` (e.g. `nytprofcalls-5.28`);
don't use scripts on PATH

Ticket: https://trac.macports.org/ticket/56635

diff --git bin/nytprofhtml bin/nytprofhtml
index a47bfbc..d9bb167 100755
--- bin/nytprofhtml
+++ bin/nytprofhtml
@@ -72,16 +72,14 @@ my $has_json = eval { require JSON::MaybeXS; JSON::MaybeXS->import(); 1 }
 
 my $script_ext   = ($^O eq "MSWin32") ? "" : ".pl";
 
-my $nytprofcalls = File::Spec->catfile($Config{'bin'}, 'nytprofcalls');
-$nytprofcalls    = which 'nytprofcalls' if not -e $nytprofcalls;
+my $nytprofcalls = File::Spec->catfile($Config{'bin'}, 'nytprofcalls-__MP_PERL5_MAJOR__');
 
-die "Unable to find nytprofcalls in $Config{bin} or on the PATH"
+die "Unable to find nytprofcalls in $Config{bin}"
     unless $nytprofcalls;
 
-my $flamegraph   = File::Spec->catfile($Config{'bin'}, 'flamegraph') . $script_ext;
-$flamegraph      = which "flamegraph$script_ext" if not -e $flamegraph;
+my $flamegraph   = File::Spec->catfile($Config{'bin'}, 'flamegraph') . $script_ext . '-__MP_PERL5_MAJOR__';
 
-die "Unable to find flamegraph$script_ext in $Config{bin} or on the PATH"
+die "Unable to find flamegraph$script_ext-__MP_PERL5_MAJOR__ in $Config{bin}"
     unless $flamegraph;
 
 my @treemap_colors = (0,2,4,6,8,10,1,3,5,7,9);
@@ -116,7 +114,7 @@ DB::set_option('blocks', 0) if $opt_minimal;
 
 sub usage {
     print <<END;
-usage: [perl] nytprofhtml [opts]
+usage: nytprofhtml-__MP_PERL5_MAJOR__ [opts]
  --file <file>, -f <file>  Read profile data from the specified file [default: nytprof.out]
  --out <dir>,   -o <dir>   Write report files to this directory [default: nytprof]
  --delete,      -d         Delete any old report files in <dir> first
